### PR TITLE
Implement Notification channel API methods

### PR DESCRIPTION
**Pull request by stefafafan**
I implemented methods that correspond to the APIs listed here: https://mackerel.io/api-docs/entry/channels

- GET `/api/v0/channels`
- POST `/api/v0/channels`
- DELETE `/api/v0/channels/<channelId>`

### CODE REVIEW

File: channels.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+// Channel represents a Mackerel notification channel.
+// ref. https://mackerel.io/api-docs/entry/channels
+type Channel struct {
+	ID string `json:"id"`
+	ChannelWithoutID
+}
+
+// ChannelWithoutID represents a Mackerel notification channel without the ID.
+type ChannelWithoutID struct {
```
Comment (lufia) [inline channels.go]: I understand this type is used for *CreateChannel*. But the type named XxxWithoutID is introduced for the first time at this p-r.

So I suggest to add `omitempty` tag to *Channel.ID*, or rename *ChannelWithoutID* to *CreateChannelParams*, like *CreateDashboard* or *CreateHost*.

File: channels.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+// Mentions represents the structure used for slack channel mentions
+type Mentions struct {
+	OK       string `json:"ok,omitempty"`
+	Warning  string `json:"warning,omitempty"`
+	Critical string `json:"critical,omitempty"`
+}
+
+// ListChannels requests the channels API and returns a list of Channel
+func (c *Client) ListChannels() ([]*Channel, error) {
```
Comment (lufia) [inline channels.go]: In mackerel-client-go, almost GET-ish method names is prefixed *Find*, except *ListHostMetricNames* and *ListServiceMetricNames*. Would you consider whether to rename?

File: channels.go (hunk 1 above)
Comment (stefafafan) [reply channels.go]: Aha, I see.
I feel like using `omitempty` that way is not accurate so I think I will change it to `CreateChannelParams`

File: channels.go (hunk 2 above)
Comment (stefafafan) [reply channels.go]: Ah. Personally, I feel `Find...` suggests that it only returns 1 item, and `List...` returns an array of items. However, it is better to be consistent so I will accept your suggestion. Thanks!

**Comment by stefafafan** [comment]
@lufia Thank you very much for the review.
I accepted your suggestions and added commits to this pull request.

**Comment by lufia** [comment]
@stefafafan Sorry, I was wrong. I think `type Channel` shouldn't embed `CreateChannelParams`, and I want you to redefine , like this.

 ```go
type Channel struct {
    ID string
    Name string
    Type string
}
```

In Go, the embedding is same as to omit the field name.

```go
type Channel struct {
    ID string
    CreateChannelParams
}
```

The same as(roughly):

```go
type Channel struct {
    ID string
    CreateChannelParams CreateChannelParams
}
```
Thus a user should assign some fields of embedded type to Channel.CreateChannelParams when it is initialized at once.

https://play.golang.org/p/tY2sxGN-I4x

**Comment by stefafafan** [comment]
@lufia Thank you pointing that out! Nice catch.
I decided to stop making a separate struct, and made `Channel.ID` omitempty so that it could be used for CreateChannel as well.

---

Other than this, I noticed that `EnabledGraphImage` will not work as expected when initializing with false. This is because omitempty omits the field when the field is falsy.

>

**Comment by stefafafan** [comment]
> The Mackerel document does not specify what will happen if you do not have this field when creating a Slack channel, but I think this field should not be omitted from the request.

I tried posting to the API with cURL, and the following error was returned.

`Invalid Parameter: obj 'enabledGraphImage' is undefined on object`

This was the same for the other fields that could accidentally get omitted (like giving an empty array for `Emails` , `UserIDs` , or `Events`), so I am considering making those pointer to arrays maybe...

**Pull request merged**